pkg/cmdutil/cmdutil.go

```diff
@@ -134,3 +135,27 @@ func FilterValidKafkas(f *factory.Factory, toComplete string) (validNames []stri
 
 	return validNames, directive
 }
+
+// FetchCloudProviders returns the list of supported cloud providers for creating a Kafka instance
+// This is used in the cmd.RegisterFlagCompletionFunc for dynamic completion of --provider
+func FetchCloudProviders(f *factory.Factory) (validProviders []string, directive cobra.ShellCompDirective) {
+	validProviders = []string{}
+	directive = cobra.ShellCompDirectiveNoSpace
+
+	conn, err := f.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return validProviders, directive
+	}
+
+	api := conn.API()
+
+	cloudProviderResponse, _, err := api.Kafka().GetCloudProviders(context.Background()).Execute()
```

Comment:
I think we should start chaining the entire conn like this where possible:

```suggestion
	cloudProviderResponse, _, err := conn.API().Kafka().GetCloudProviders(context.Background()).Execute()
```